pkg/shardddl/optimism/lock.go

```diff
@@ -214,7 +219,12 @@ func (l *Lock) TryRemoveTable(source, schema, table string) bool {
 		return false
 	}
 
-	delete(l.tables[source][schema], table)
+	if ti, ok := l.tables[source][schema][table]; ok {
```

Comment:
why add this judgment? seems that L218 already do this

Reply:
My mistake, fixed in https://github.com/pingcap/dm/pull/615/commits/41085a7905db8ab74bde7030c91ae868e9c2f370